### ISSUE TITLE
One of 3 tikv nodes suddenly transitioned to Pending Offline state, how to resolve it?

### ISSUE THREAD

**Issue by uqix** [pingcap/tidb#23446]
```
$ tiup cluster display tidb-test
Starting component `cluster`:  display tidb-test
Cluster type:    tidb
Cluster name:    tidb-test
Cluster version: v4.0.8
SSH type:        builtin
ID                Role          Host        Ports        OS/Arch       Status           Data Dir                           Deploy Dir
--                ----          ----        -----        -------       ------           --------                           ----------
172.16.0.5:9093   alertmanager  172.16.0.5  9093/9094    linux/x86_64  Up               /data/tidb-data/alertmanager-9093  /data/tidb-deploy/alertmanager-9093
172.16.0.5:3000   grafana       172.16.0.5  3000         linux/x86_64  Up               -                                  /data/tidb-deploy/grafana-3000
172.16.0.5:2379   pd            172.16.0.5  2379/2380    linux/x86_64  Up|L|UI          /data/tidb-data/pd-2379            /data/tidb-deploy/pd-2379
172.16.0.8:2379   pd            172.16.0.8  2379/2380    linux/x86_64  Up               /data/tidb-data/pd-2379            /data/tidb-deploy/pd-2379
172.16.0.5:9090   prometheus    172.16.0.5  9090         linux/x86_64  Up               /data/tidb-data/prometheus-9090    /data/tidb-deploy/prometheus-9090
172.16.0.5:4000   tidb          172.16.0.5  4000/10080   linux/x86_64  Up               -                                  /data/tidb-deploy/tidb-4000
172.16.0.8:4000   tidb          172.16.0.8  4000/10080   linux/x86_64  Up               -                                  /data/tidb-deploy/tidb-4000
172.16.0.2:20160  tikv          172.16.0.2  20160/20180  linux/x86_64  Pending Offline  /data/tidb-data/tikv-20160         /data/tidb-deploy/tikv-20160
172.16.0.4:20160  tikv          172.16.0.4  20160/20180  linux/x86_64  Up               /data/tidb-data/tikv-20160         /data/tidb-deploy/tikv-20160
172.16.0.6:20160  tikv          172.16.0.6  20160/20180  linux/x86_64  Up               /data/tidb-data/tikv-20160         /data/tidb-deploy/tikv-20160
Total nodes: 10
```

So far, I have tried something like restarting the cluster, scale-in then scale-out the `Pending Offline` node, nothing worked.

pd logs:
```
[2021/03/20 14:40:44.601 +08:00] [WARN] [cluster.go:1147] ["store may not turn into Tombstone, there are no extra up store has enough space to accommodate the extra replica"] [store="id:1 address:\"172.16.0.2:20160\" state:Offline version:\"4.0.8\" status_address:\"0.0.0.0:20180\" git_hash:\"83091173e960e5a0f5f417e921a0801d2f6635ae\" start_timestamp:1616221816 deploy_path:\"/data/tidb-deploy/tikv-20160/bin\" last_heartbeat:1616222438019455417 "]
```

tikv logs:
```
[2021/03/20 14:48:05.069 +08:00] [INFO] [raft.rs:285] [newRaft] [peers="[(78072, Progress { matched: 0, next_idx: 6, state: Probe, paused: false, pending_snapshot: 0, pending_request_snapshot: 0, recent_active: false, ins: Inflights { start: 0, count: 0, buffer: [] } }), (78070, Progress { matched: 5, next_idx: 6, state: Probe, paused: false, pending_snapshot: 0, pending_request_snapshot: 0, recent_active: false, ins: Inflights { start: 0, count: 0, buffer: [] } }), (78071, Progress { matched: 0, next_idx: 6, state: Probe, paused: false, pending_snapshot: 0, pending_request_snapshot: 0, recent_active: false, ins: Inflights { start: 0, count: 0, buffer: [] } })]"] ["last term"=5] ["last index"=5] [applied=5] [commit=5] [term=5] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:05.069 +08:00] [INFO] [raw_node.rs:222] ["RawNode created with id 78070."] [id=78070] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:05.069 +08:00] [INFO] [raft.rs:1192] ["[logterm: 5, index: 5, vote: 0] cast vote for 78072 [logterm: 5, index: 5] at term 5"] ["msg type"=MsgRequestPreVote] [term=5] [msg_index=5] [msg_term=5] [from=78072] [vote=0] [log_index=5] [log_term=5] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:05.072 +08:00] [INFO] [raft.rs:1003] ["received a message with higher term from 78072"] ["msg type"=MsgRequestVote] [message_term=6] [term=5] [from=78072] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:05.072 +08:00] [INFO] [raft.rs:783] ["became follower at term 6"] [term=6] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:05.072 +08:00] [INFO] [raft.rs:1192] ["[logterm: 5, index: 5, vote: 0] cast vote for 78072 [logterm: 5, index: 5] at term 6"] ["msg type"=MsgRequestVote] [term=6] [msg_index=5] [msg_term=5] [from=78072] [vote=0] [log_index=5] [log_term=5] [raft_id=78070] [region_id=78069]
[2021/03/20 14:48:35.798 +08:00] [INFO] [raft.rs:1003] ["received a message with higher term from 1303"] ["msg type"=MsgRequestVote] [message_term=120679] [term=120678] [from=1303] [raft_id=1302] [region_id=1301]
[2021/03/20 14:48:35.798 +08:00] [INFO] [raft.rs:783] ["became follower at term 120679"] [term=120679] [raft_id=1302] [region_id=1301]
[2021/03/20 14:48:35.798 +08:00] [INFO] [raft.rs:1192] ["[logterm: 120678, index: 26312656, vote: 0] cast vote for 1303 [logterm: 120678, index: 26312656] at term 120679"] ["msg type"=MsgRequestVote] [term=120679] [msg_index=26312656] [msg_term=120678] [from=1303] [vote=0] [log_index=26312656] [log_term=120678] [raft_id=1302] [region_id=1301]
[2021/03/20 14:48:49.367 +08:00] [INFO] [gc_manager.rs:416] ["gc_worker: start auto gc"] [safe_point=423682951912095744]
```

**Comment by uqix** [comment]
Or how to cancel the scale-in operation considering a cluster needs at least 3 tikv nodes?

**Comment by ljluestc** [comment]
```

It seems like the TiKV node with the address `172.16.0.2:20160` is still in a pending offline state despite attempting various solutions. 

Given that the issue persists, let's try a few additional troubleshooting steps:

1. **Verify connectivity**: Ensure that there are no network issues preventing communication between the TiKV node and other components of the TiDB cluster.

2. **Check disk space**: Double-check the available disk space on the TiKV node (`172.16.0.2`) to confirm that it has sufficient space for the TiKV data directory.

3. **Inspect TiKV logs**: Review the TiKV logs again to see if there are any new errors or warnings that could provide further insights into the root cause of the issue.

4. **Manual intervention**: If all else fails, you may need to manually intervene by removing the TiKV instance from the cluster and then adding it back. This involves stopping TiKV on the affected node, removing its data directory, and then adding it back to the TiDB cluster.

If you've already tried these steps or need assistance with any specific aspect, feel free to let me know, and I can provide further guidance.
```

**Comment by jebter** [comment]
The node is being scaled in, but there are not enough three nodes to scale in. I will close this issue. If there are any updates, you can reopen it.
